okhttp/src/main/kotlin/okhttp3/internal/cache/DiskLruCache.kt

```diff
@@ -90,6 +102,18 @@ class DiskLruCache(
   /** Used for asynchronous journal rebuilds. */
   taskRunner: TaskRunner
 ) : Closeable, Flushable {
+  internal val fileSystem: FileSystem = object : ForwardingFileSystem(fileSystem) {
+    override fun sink(file: Path): Sink {
+      file.parent?.let {
+        // TODO from okhttp3.internal.io.FileSystem
```

Comment:
This is good because it’s directly equivalent. In follow up I think we should do this once in the init function.